hazelcast/src/main/java/com/hazelcast/jet/impl/SubmitJobParameters.java

```diff
@@ -14,23 +14,28 @@
  * limitations under the License.
  */
 
-package com.hazelcast.jet;
+package com.hazelcast.jet.impl;
 
 import javax.annotation.Nonnull;
 import java.nio.file.Path;
 import java.util.Collections;
 import java.util.List;
 
 /**
- * The parameter object for {{@link JetService#submitJobFromJar(SubmitJobParameters)}}
+ * The parameter object for {{@link JetClientInstanceImpl#submitJobFromJar(SubmitJobParameters)}}
  */
-public class SubmitJobParameters {
+public final class SubmitJobParameters {
 
     /**
      * Path of the jar
      */
     private Path jarPath;
 
+    /**
+     * Set to true if the jar is already present on the member. The jar will be executed directly
+     */
+    private boolean directJobExecution;
```

Comment:
What does `executed directly` mean? I think a better name would be something like `jarOnMember`, and then
- forJobUpload -> withJarOnClient - sets `jarOnMember` to false
- forDirectJobExecution -> withJarOnMember - sets `jarOnMember to true

Reply:
jarOnMember implies that the jar resides on the member. However we don't care where the jar resides. It can be on another  POD, on database or on a shared volume. It is only accessible to member.  We only want to execute it without upload.

Reply:
>jarOnMember implies that the jar resides on the member

yes, exactly, we don't have any other hypothetical source for the jars atm. neither we plan to have.

Reply:
Rev : e80ca67d6b7e8438894d84e92736850a99de53a9 and e2cc2596bbffc425558aa2306cda0b7754cf9106